app/models/alchemy/element.rb

```diff
@@ -321,9 +321,11 @@ def copy_nested_elements_to(target_element)
     #   Pass an element name to get previous of this kind.
     #
     def previous_or_next(dir, name = nil)
-      elements = page.elements.published.where("#{self.class.table_name}.position #{dir} #{position}")
+      dir = dir == '>' ? '>' : '<'
+      order = dir == '>' ? :asc : :desc
+      elements = page.elements.published.where("#{self.class.table_name}.position #{dir} ?", position)
```

Comment:
Also, is there a reason why the query interpolates `self.class.table_name` in the query?

Would we be able to rewrite the query as something like:

```ruby
page.elements.published.where('position < ?' position)
```

Then we could just rely on ActiveRecord to produce the appropriate SQL.